src/eventListeners/onImageRenderedBrushEventHandler.js

```diff
@@ -1,263 +1,857 @@
 import store from '../store/index.js';
-import getActiveToolsForElement from '../store/getActiveToolsForElement.js';
-import { getToolState } from '../stateManagement/toolState.js';
 import external from '../externalModules.js';
-import BaseBrushTool from './../tools/base/BaseBrushTool.js';
 import {
   getNewContext,
-  resetCanvasContextTransform,
-  transformCanvasContext,
+  draw,
+  drawLines,
+  drawJoinedLines,
 } from '../drawing/index.js';
 
-/* Safari and Edge polyfill for createImageBitmap
- * https://developer.mozilla.org/en-US/docs/Web/API/WindowOrWorkerGlobalScope/createImageBitmap
+import { getLogger } from '../util/logger.js';
+
+const logger = getLogger('eventListeners:onImageRenderedBrushEventHandler');
+
+const { state, configuration, getters } = store.modules.segmentation;
+
+/**
+ * Renders segmentation labelmap data associated with this element, with
+ * settings defined in the segmentationModules configuration.
+ *
+ * @param {Object} evt - The cornerstone event.
+ * @returns {null}
  */
+export default function(evt) {
+  const eventData = evt.detail;
+  const element = eventData.element;
 
-// TODO: Do we still need this? I've yanked the package for now
-// It should be covered by @babel/runtime and plugin-transform-runtime:
-// https://babeljs.io/docs/en/babel-plugin-transform-runtime
-// @James, I think Babel should take care of this for us
-// Import regeneratorRuntime from "regenerator-runtime";
+  const {
+    activeLabelmapIndex,
+    labelmaps3D,
+    currentImageIdIndex,
+  } = getters.labelmaps3D(element);
 
-if (!('createImageBitmap' in window)) {
-  window.createImageBitmap = function(imageData) {
-    return new Promise(resolve => {
-      const img = document.createElement('img');
+  if (!labelmaps3D) {
+    return;
+  }
 
-      img.addEventListener('load', function() {
-        resolve(this);
-      });
+  if (configuration.renderInactiveLabelmaps) {
```

Comment:
`shouldRenderInactiveLabelMaps` (note `LabelMaps` I notice in some places we use that capitalization, in others it's `Labelmaps`)